hub/integrations/pytorch.py

```diff
@@ -57,19 +62,38 @@ def _read_and_store_chunk(
     return chunk_size
 
 
-def dataset_to_pytorch(dataset, transform: Callable = None, workers: int = 1):
+def dataset_to_pytorch(
+    dataset,
+    transform: Optional[Callable] = None,
+    workers: int = 1,
+    tuple_fields: Optional[List] = None,
+):
     dataset.flush()
-    return TorchDataset(dataset, transform, workers)
+    return TorchDataset(dataset, transform, workers, tuple_fields)
 
 
 class TorchDataset:
-    def __init__(self, dataset, transform: Callable = None, workers: int = 1):
+    def __init__(
+        self,
+        dataset,
+        transform: Optional[Callable] = None,
+        workers: int = 1,
+        tuple_fields: Optional[List[str]] = None,
+    ):
         self._import_torch()
         self.transform: Optional[Callable] = transform
         self.workers: int = workers
+        self.tuple_fields = tuple_fields
         self.map = ProcessPool(nodes=workers).map
         self.length = len(dataset)
-        self.keys = list(dataset.tensors)
+        self.keys: List[str] = list(dataset.tensors)
```

Comment:
I think we can simplify this slightly and remove self.tuple_fields by only using self.keys and assigning tuple_fields to to it if it is passed. This will also improve other functions like load_all_index_meta and load_all_meta as they won't fetch extra data for keys that are not needed.